pybamm/models/base_model.py

```diff
@@ -936,6 +942,341 @@ def default_solver(self):
+
+        return geo
+
+    def _get_geometry_displays(self, var):
+        """
+        Returns min range from the first domain and max range from the last domain of
+        all nodes in latex.
+        """
+        geo = []
+
+        if not var.domain:
+            return geo
+
+        rng_min = None
+        rng_max = None
+        name = None
+
+        # Take range minimum from the first domain
+        for var_name, rng in self.default_geometry[var.domain[0]].items():
+            # Trim name (r_n --> r)
+            name = re.findall(r"(.)_*.*", str(var_name))[0]
+            if getattr(rng["min"], "print_name", None) is None:
+                rng_min = rng["min"]
+            else:
+                rng_min = rng["min"].print_name
+
+        # Take range maximum from the last domain
+        for var_name, rng in self.default_geometry[var.domain[-1]].items():
+            if getattr(rng["max"], "print_name", None) is None:
+                rng_max = rng["max"]
+            else:
+                rng_max = rng["max"].print_name
+
+        geo_latex = f"\quad {rng_min} < {name} < {rng_max}"
+        geo.append(geo_latex)
+
+        return geo
+
+    def _get_bcs_displays(self, lhs_dr, var):
+        """
+        Returns a list of boundary condition equations with ranges in front of
+        the equations.
+        """
+        bcs_eqn_list = []
+        bcs = self.boundary_conditions.get(var, None)
+
+        if bcs:
+            # Take range minimum from the first domain
+            for var_name, rng in self.default_geometry[var.domain[0]].items():
+                # Trim name (r_n --> r)
+                name = re.findall(r"(.)_*.*", str(var_name))[0]
+
+                if getattr(rng["min"], "print_name", None) is None:
+                    rng_min = rng["min"]
+                else:
+                    rng_min = rng["min"].print_name
+
+                bcs_left = sympy.latex(bcs["left"][0].to_equation())
+                bcs_left_latex = bcs_left + f"\quad {name} = {rng_min}"
+                bcs_eqn = sympy.Eq(lhs_dr, sympy.Symbol(bcs_left_latex), evaluate=False)
+                bcs_eqn_list.append(bcs_eqn)
+
+            # Take range maximum from the last domain
+            for var_name, rng in self.default_geometry[var.domain[-1]].items():
+                # Trim name (r_n --> r)
+                name = re.findall(r"(.)_*.*", str(var_name))[0]
+
+                if getattr(rng["max"], "print_name", None) is None:
+                    rng_max = rng["max"]
+                else:
+                    rng_max = rng["max"].print_name
+
+                bcs_right = sympy.latex(bcs["right"][0].to_equation())
+                bcs_right_latex = bcs_right + f"\quad {name} = {rng_max}"
+                bcs_eqn = sympy.Eq(
+                    lhs_dr, sympy.Symbol(bcs_right_latex), evaluate=False
+                )
+                bcs_eqn_list.append(bcs_eqn)
+
+        return bcs_eqn_list
+
+    def _get_names(self, node):
```

Comment:
It's a bit weird that this is a function in `base_model` - it doesn't use `self`